### DISCUSSION TITLE
How to import todoist tasks using task creation date ?

### DISCUSSION THREAD

**Discussion by mathisgauthey** [chhoumann/quickadd#554] [Q&A]
Hi everyone, I recently came across your [guide](https://quickadd.obsidian.guide/docs/Examples/Capture_FetchTasksFromTodoist) to import my Todoist Tasks inside Obsidian.

Is there a way to import the todoist task creation date to import them in the following format :
`- [ ] task.content ➕ task.creation_date 📅 task.due_date`

With the last one being optional.

I managed to make it work using the actual date, but I'd like to use the task property to be able to do it on the next day.

```
function formatTasksToTasksPluginTask(t) {
  const todayDate = new Date().toISOString().split("T")[0]; // Get today's date in YYYY-MM-DD format
  return (
    t
      .map(
        (t) =>
          (t = t.rawDatetime
            ? `- [ ] ${t.content} \u2795 ${todayDate} \ud83d\udcc5 ${t.rawDatetime.format(
                "YYYY-MM-DD"
              )}`
            : `- [ ] ${t.content} \u2795 ${todayDate}`)
      )
      .join("\n") + "\n"
  );
}
```

**Comment by chhoumann** [comment]
Coming back to this far too late - sorry.

The tasks in that example come through the Todoist plugin's own API (`api.getTasks()`), so the fields you can use are whatever that plugin puts on its task objects - the example script only touches `content` and `rawDatetime`. Todoist itself stores a creation timestamp on every task, so step one is to check whether it survives the plugin wrapper: drop a `console.log(tasks[0])` into `getAllTasks` and look in the developer console (Ctrl+Shift+I) for a field like `created_at` (or `added_at`).

If it's there, your function only needs the one extra mapping:

```js
function formatTasksToTasksPluginTask(t) {
  return (
    t
      .map((t) => {
        const created = window.moment(t.created_at).format("YYYY-MM-DD");
        return t.rawDatetime
          ? `- [ ] ${t.content} \u2795 ${created} \ud83d\udcc5 ${t.rawDatetime.format("YYYY-MM-DD")}`
          : `- [ ] ${t.content} \u2795 ${created}`;
      })
      .join("\n") + "\n"
  );
}
```

If the plugin strips the field, the fallback is to fetch tasks yourself from Todoist's REST API with your API token instead of going through the plugin - the raw task objects there always carry the creation timestamp.

If you log a task and paste what fields it has, I can pin down the exact property name for you.
